fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -75,6 +108,57 @@ def fit(self, X, y, aux_data=None, **kwargs):
         # Prep the quality metric
         self.quality_metric.set_data(X, y_vector, A)
 
+        if isinstance(self.disparity_metric, Moment):
+            # Prep the disparity metric and objective
+            self.disparity_metric.init(X, A, y_vector)
+            objective = self.disparity_metric.default_objective()
+            objective.init(X, A, y_vector)
+            is_classification_reduction = (self.disparity_metric.reduction_type == _REDUCTION_TYPE_CLASSIFICATION)
+
+            # Basis information
+            pos_basis = self.disparity_metric.pos_basis
+            neg_basis = self.disparity_metric.neg_basis
+            neg_allowed = self.disparity_metric.neg_basis_present
+            objective_in_the_span = (self.disparity_metric.default_objective_lambda_vec is not None)
+            dim = len(pos_basis.columns)
+
+            # Grid parameters: these should be provided as arguments, but explicitly here for the time being
+            grid_limit = 2.0
+            grid_size = number_of_lagrange_multipliers-1
+
+            # Generate the grid
+            grid = _GridGenerator(grid_limit, grid_size, dim, neg_allowed, objective_in_the_span).grid
+
+            # Fit the estimates
+            self.all_results = []
+            for i in grid:
+                beta_pos = grid[i].copy()
+                beta_neg = -grid[i].copy()
+                beta_pos[beta_pos<0] = 0.0
+                beta_neg[beta_neg<0] = 0.0
+                lambda_vec = pos_basis.dot(beta_pos) + neg_basis.dot(beta_neg)
+                weights = self.disparity_metric.signed_weights(lambda_vec)
+                if not objective_in_the_span:
+                    weights = weights + objective.signed_weights()
+                if is_classification_reduction:
+                    y_reduction = 1 * (weights > 0)
+                    weights = weights.abs()
+                else:
+                    y_reduction = y
+
+                current_learner = copy.deepcopy(self.learner)
+                current_learner.fit(X, y_reduction, sample_weight=weights)
+
+                # Evaluate the quality metric
+                quality = self.quality_metric.get_quality(current_learner)
```

Comment:
Do we want to keep a separate `quality_metric` or is this just a convenience for now, so that not too much breaks at once?